fog/view/enclave/impl/src/oblivious_utils.rs

```diff
@@ -0,0 +1,627 @@
+        )
+        .into();
+        assert!(result);
+
+        shard_tx_out_search_result = create_test_tx_out_search_result(
+            search_key,
+            0,
+            FIXED_CIPHERTEXT_LENGTH - 1,
+            TxOutSearchResultCode::RateLimited,
+        );
+        result = should_overwrite_tx_out_search_result(
+            &client_tx_out_search_result,
+            &shard_tx_out_search_result,
+        )
+        .into();
+        assert!(result);
+    }
+
+    #[test]
+    fn should_overwrite_tx_out_search_result_client_has_bad_search_key_shard_has_retryable_error_returns_false(
+    ) {
+        let search_key = vec![0u8; 10];
+        let client_tx_out_search_result = create_test_tx_out_search_result(
+            search_key.clone(),
+            0,
+            FIXED_CIPHERTEXT_LENGTH - 1,
+            TxOutSearchResultCode::BadSearchKey,
+        );
+
+        let mut shard_tx_out_search_result = create_test_tx_out_search_result(
+            search_key.clone(),
+            0,
+            FIXED_CIPHERTEXT_LENGTH - 1,
+            TxOutSearchResultCode::InternalError,
+        );
+        let mut result: bool = should_overwrite_tx_out_search_result(
+            &client_tx_out_search_result,
+            &shard_tx_out_search_result,
+        )
+        .into();
+        assert!(!result);
+
+        shard_tx_out_search_result = create_test_tx_out_search_result(
+            search_key,
+            0,
+            FIXED_CIPHERTEXT_LENGTH - 1,
+            TxOutSearchResultCode::RateLimited,
+        );
+        result = should_overwrite_tx_out_search_result(
+            &client_tx_out_search_result,
+            &shard_tx_out_search_result,
+        )
+        .into();
+        assert!(!result);
+    }
+
+    #[test]
+    fn should_overwrite_tx_out_search_result_client_has_bad_search_key() {
+        let search_key = vec![0u8; 10];
+        let shard_tx_out_search_result = create_test_tx_out_search_result(
+            search_key.clone(),
+            0,
+            FIXED_CIPHERTEXT_LENGTH - 1,
+            TxOutSearchResultCode::BadSearchKey,
+        );
+
+        let mut client_tx_out_search_result = create_test_tx_out_search_result(
+            search_key.clone(),
+            0,
+            FIXED_CIPHERTEXT_LENGTH - 1,
+            TxOutSearchResultCode::RateLimited,
+        );
+        let mut result: bool = should_overwrite_tx_out_search_result(
+            &client_tx_out_search_result,
+            &shard_tx_out_search_result,
+        )
+        .into();
+        assert!(result);
+
+        client_tx_out_search_result = create_test_tx_out_search_result(
+            search_key,
+            0,
+            FIXED_CIPHERTEXT_LENGTH - 1,
+            TxOutSearchResultCode::RateLimited,
+        );
+        result = should_overwrite_tx_out_search_result(
+            &client_tx_out_search_result,
+            &shard_tx_out_search_result,
+        )
+        .into();
+        assert!(result);
+    }
```

Comment:
The test is correct as written. The test is not supposed to test when the client result is `TxOutSearchResultCode::NotFound`, rather it tests when the code is `!TxOutSearchResultCode::Found` - i.e. any code that isn't `TxOutSearchResultCode::Found`.Since `TxOutSearchResultCode::RateLimited != TxOutSearchResultCode::Found`, the test is valid.